server/src/main/java/com/cloud/storage/snapshot/SnapshotManagerImpl.java

```diff
@@ -1332,33 +1327,12 @@ private void updateSnapshotPayload(long storagePoolId, CreateSnapshotPayload pay
         }
     }
 
-    private DataStoreRole getDataStoreRole(Snapshot snapshot, SnapshotDataStoreDao snapshotStoreDao, DataStoreManager dataStoreMgr) {
-        SnapshotDataStoreVO snapshotStore = snapshotStoreDao.findBySnapshot(snapshot.getId(), DataStoreRole.Primary);
-
-        if (snapshotStore == null) {
-            return DataStoreRole.Image;
-        }
-
-        long storagePoolId = snapshotStore.getDataStoreId();
-        DataStore dataStore = dataStoreMgr.getDataStore(storagePoolId, DataStoreRole.Primary);
-
-        Map<String, String> mapCapabilities = dataStore.getDriver().getCapabilities();
-
-        if (mapCapabilities != null) {
-            String value = mapCapabilities.get(DataStoreCapabilities.STORAGE_SYSTEM_SNAPSHOT.toString());
-            Boolean supportsStorageSystemSnapshots = new Boolean(value);
-
```

Comment:
@slavkap does the removal of below code cause regression in other storage pool types?

Reply:
@slavkap I agree, it doesn't break for the pools with this last capability flag set. Last code returns `DataStoreRole.Primary` when the snapshot exists in the primary storage and the storage system snapshot capability flag is set for the pool. No issues with 'PowerFlex', as the flag is set in _ScaleIOPrimaryDataStoreDriver_. It seems this is not set for 'RBD', Can you check if there is any issue with it and other pools using _CloudStackPrimaryDataStoreDriverImpl_. Thanks.